components/ruby/lib/chef-licensing/license_key_fetcher/file.rb

```diff
@@ -208,9 +214,23 @@ def read_license_key_file
         # only checking for major version for file format for breaking changes
         @contents ||= YAML.load(::File.read(path))
         if major_version(@contents[:file_format_version]) == major_version(LICENSE_FILE_FORMAT_VERSION)
+          current_version_class_name = Object.const_get("ChefLicensing::LicenseKeyFetcher::V#{major_version(LICENSE_FILE_FORMAT_VERSION)}LicenseFile")
+          raise LicenseFileCorrupted.new("Invalid data found in the license file.") unless current_version_class_name.send(:verify_structure, @contents)
+
+          @contents
+        elsif SUPPORTED_LICENSE_FILE_FORMAT_VERSIONS.include?(@contents[:file_format_version])
+          logger.warn "License File version #{@contents[:file_format_version]} is deprecated."
+          # TODO: Decide if we want to support migrating from older versions of the license file automatically
+          logger.warn "Automatically migrating license file to version #{LICENSE_FILE_FORMAT_VERSION}."
+          given_version_class_name = Object.const_get("ChefLicensing::LicenseKeyFetcher::V#{major_version(@contents[:file_format_version])}LicenseFile")
+          raise LicenseFileCorrupted.new("Invalid data found in the license file.") unless given_version_class_name.send(:verify_structure, @contents)
+
+          @contents = given_version_class_name.send(:migrate_structure, @contents, major_version(LICENSE_FILE_FORMAT_VERSION))
```

Comment:
per this we expect to change V3LicenseFile migrate_structure method again when we introduce V5 .. instead lets to the opposite if we find old files and we know the current upgradable version than call the migrate structure of the current version file

Reply:
cool, makes sense.